cabal-testsuite/PackageTests/HaddockKeepsTmps/cabal.test.hs

```diff
@@ -1,21 +1,33 @@
 {-# LANGUAGE LambdaCase #-}
 import Test.Cabal.Prelude
-import Data.List (sort)
+import Data.List (sort, isPrefixOf)
 import Distribution.Verbosity
 import Distribution.Simple.Glob
 import Distribution.Simple.Glob.Internal
 import Distribution.Simple.Utils
+import System.Directory (getTemporaryDirectory)
 
 -- Test that "cabal haddock" preserves temporary files
 -- We use haddock-keep-temp-file: True in the cabal.project.
-main = cabalTest $ recordMode DoNotRecord $ withProjectFile "cabal.project" $ do
-    cabal "haddock" []
-
-    cwd <- fmap testCurrentDir getTestEnv
-
-    -- Windows has multiple response files, and only the last one (alphabetically) is the important one.
-    (safeLast . sort . globMatches <$> liftIO (runDirFileGlob silent Nothing cwd (GlobDirRecursive [WildCard, Literal "txt"]))) >>= \case
-      Nothing -> error "Expecting a response file to exist"
+main = do
+  cwd <- getTemporaryDirectory
+  cabalTest $ recordMode DoNotRecord $ withProjectFile "cabal.project" $ do
+    res <- cabal' "haddock" []
+    -- The output will show something like:
+    -- > Running: ...haddock... '@<file>'
+    -- And we want to get the <file>
+    let rspFile = do
+          -- If there are multiple, we only care about one of them
+          aHaddockInvocation <- safeHead [ l | l <- lines (resultOutput res),
+                                           "Running: " `isPrefixOf` l,
+                                           "haddock"   `isInfixOf`  l,
+                                           '@'         `elem`       l
+                                         ]
+          -- This one is surrounded by backticks and has a '@' at the beginning
+          theQuotedRspFile <- safeHead $ drop 2 $ words aHaddockInvocation
+          pure $ init $ drop 2 $ theQuotedRspFile
```

Comment:
I don't really like how this test is structured now. If you want to assert anything about the temporary files created then you have to print out the path into the log in order to be able to find them.

Could we not instead set `TMPDIR` when running tests to place the files into a temporary directory inside the temporary directory where the test is already running? Then it is easy to query this to write tests to see what is actually created and left over.

Reply:
I reworked the test to point to a temp directory specific for this test.